### PR TITLE
fix coinbase in auto exec

### PR DESCRIPTION
**Pull request by ouwenkg**
# fix coinbase in auto exec

### Description of the Change

In the case of auto exec situation, coinbase should be proposer.

### Applicable Issues
#493

### CODE REVIEW

File: cita-executor/core/src/libexecutor/auto_exec.rs [hunk 1]
```diff
@@ -68,7 +69,7 @@ pub fn auto_exec(
     let mut trace_output = tracer.prepare_trace_output();
     let output = OutputPolicy::Return(BytesRef::Flexible(&mut out), trace_output.as_mut());
     let factory = Factory::new(VMType::Interpreter, 1024 * 32);
-    let env_info = EnvInfo::default();
```
Comment (kaikai1024) [inline cita-executor/core/src/libexecutor/auto_exec.rs]: Just delete it.

File: cita-executor/core/src/libexecutor/block.rs [hunk 2]
```diff
@@ -208,6 +206,9 @@ impl ExecutedBlock {
                 b
             });
 
+        block.set_state_root(state_root);
```
Comment (kaikai1024) [inline cita-executor/core/src/libexecutor/block.rs]: It's nothing about the commit.
A new commit is better.

Actually it's not a good habit to do things beyond the pr's purpose.
I've done this before, too. Maybe we all should follow it and review this each other.

Comment (kaikai1024) [reply cita-executor/core/src/libexecutor/block.rs]: BTW, I'll write the doc about the rule: PR guide, review guide, and something like this.

File: cita-executor/core/src/libexecutor/auto_exec.rs [hunk 3]
```diff
@@ -41,11 +41,12 @@ lazy_static! {
     static ref AUTO_EXEC_HASH: Vec<u8> = method_tools::encode_to_vec(AUTO_EXEC);
 }
 
-// pub fn auto_exec(state: &mut State<StateDB>) -> evm::Result<FinalizationResult> {
 pub fn auto_exec(
     state: &mut State<StateDB>,
     auto_exec_quota_limit: u64,
     economical_model: EconomicalModel,
+    // TODO: UPGRADE PROTOCAL VERSION
```
Comment (kaikai1024) [inline cita-executor/core/src/libexecutor/auto_exec.rs]: `PROTOCAL` -> `PROTOCOL`

File: cita-executor/core/src/libexecutor/auto_exec.rs (hunk 1 above)
Comment (ouwenkg) [reply cita-executor/core/src/libexecutor/auto_exec.rs]: Done!

File: cita-executor/core/src/libexecutor/auto_exec.rs (hunk 3 above)
Comment (ouwenkg) [reply cita-executor/core/src/libexecutor/auto_exec.rs]: Deal!

File: cita-executor/core/src/libexecutor/auto_exec.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-executor/core/src/libexecutor/auto_exec.rs]: I mean the blank line.
That's ok.

**Pull request merged**